discojs/src/validator.ts

```diff
@@ -14,14 +14,14 @@ export class Validator<D extends DataType> {
   /** infer every line of the dataset and check that it is as labelled */
   async *test(
     dataset: Dataset<DataFormat.Raw[D]>,
-  ): AsyncGenerator<boolean, void> {
+  ): AsyncGenerator<{ result: boolean; predicted: DataFormat.ModelEncoded[D][1]; truth : DataFormat.ModelEncoded[D][1] }, void> {
```

Comment:
Didn't expect it to be such a big refractor, because it makes the logic really different about how to compute the matrix (and at this point I am happy to have left the labels, because now I use them and not the numbers anymore)